docs: add user stories for the Menubar and SideNavigation components (#1350)
momentum-design/momentum-design: packages/components/src/components/sidenavigation/sidenavigation.mdx

import { Meta, Title, Primary, Controls } from '@storybook/blocks';

import SideNavigationStories from './sidenavigation.stories';

<Meta of={SideNavigationStories} name="User Stories" />

## Feature: SideNavigation Accessibility and User Interaction

### ✅ Background

> The sidenavigation is rendered as a persistent vertical navigation area on the left side of the page.  
> It has `role="navigation"` with accessible name (`aria-label` or `aria-labelledby`).  

> > Inside it, one or more Menubars (`role="menubar"`, `aria-orientation="vertical"`) are used.

Reviewer: Let's mention in the background section about the definition of a navigation item == menuitem

> And the arrow (collapse) button is reachable via Tab  
> When I focus the arrow button using keyboard  
> And I press the "Enter" or "Space" key  
> Then the sidenavigation collapses and displays only icons  
> And focus remains on the arrow button after collapsing

**Scenario: Expand the sidenavigation using mouse**  
> Given the sidenavigation is collapsed  
> And the expand button is visible and enabled  
> When I click on the arrow (expand) button with the mouse  
> Then the sidenavigation expands and shows labels next to icons

**Scenario: Expand the sidenavigation using keyboard**  
> Given the sidenavigation is collapsed  
> And the arrow (expand) button is reachable via Tab  
> When I focus the arrow button using keyboard  
> And I press the "Enter" or "Space" key  
> Then the sidenavigation expands and shows labels next to icons  
> And focus remains on the arrow button after expanding

---

### ✅ Scroll Behavior

**Scenario: Scroll to reveal more top navigation items using mouse or touch**  
> Given the sidenavigation contains more navigation items than fit vertically  
> And a fixed section exists at the bottom containing navigation items  

> > When I scroll using mouse wheel or swipe gesture  

Reviewer: Lets remove touch and swipe gesture

**Feature: Selecting a Top-Level Navigation Item**

**Scenario: Navigating to a new page by mouse click**  
> Given the sidenavigation is visible  
> And a navigation item without submenu is visible  
> When I click that navigation item  
> Then page navigation occurs  
> And that item receives `aria-current="page"`  
> And any previously selected navigation item loses `aria-current`

**Scenario: Navigating to a new page by keyboard**  
> Given the sidenavigation is visible  
> And focus is on a navigation item without submenu  
> When I press "Enter" or "Space"  
> Then page navigation occurs  
> And that item receives `aria-current="page"`  
> And any previously selected navigation item loses `aria-current`

> **Scenario: Triggering an inline action by clicking a navigation item**  

Reviewer: lets rename it from 'inline action' to 'action'

> Given the sidenavigation is visible  

> > And a navigation item performs an inline action (e.g., opens dialog, alert)  

Reviewer: ```suggestion
> And a navigation item can perform an action (e.g., opens dialog, alert)  
```

> When I click that navigation item  

> > Then its inline action executes  

Reviewer: ```suggestion
> Then the corresponding action is performed
```

> And no page navigation occurs  
> And `aria-current` remains unchanged

**Scenario: Triggering an inline action via keyboard**  
> Given the sidenavigation is visible  
> And focus is on an actionable navigation item  
> When I press "Enter" or "Space"  
> Then its inline action executes  
> And `aria-current` remains unchanged  

> > And focus remains in sidenavigation

Reviewer: ```suggestion
> And focus remains on the on this navigation item in sidenavigation
```

---

**Feature: Selecting a Navigation Item Inside a Submenu**

**Scenario: Selecting a nested navigation item with mouse or keyboard**  
> Given the sidenavigation is visible  

> > And the parent navigation item's submenu is open  

Reviewer: Lets create a background for these scenarios.

> And a nested navigation item is visible and focusable  
> When I activate that nested navigation item (via mouse or keyboard)  
> Then page navigation occurs  
> And that nested item receives `aria-current="page"`  
> And any previously selected navigation item loses `aria-current`  
> And the submenu closes  

> > And if activated by keyboard, focus returns to parent navigation item

Reviewer: we also need to mention that all the parent navigation item's style would change to active.

**Scenario: Re-engaging with active top-level navigation item**  
> Given a nested navigation item was previously selected  
> And the submenu is currently closed  
> And parent navigation item visually reflects active state (due to child holding `aria-current`)  
> When I hover over the parent navigation item  
> Then a tooltip displays indicating an active nested navigation item exists

> When I interact with the parent navigation item again (via mouse click or keyboard)  
> Then the submenu reopens  
> And previously selected nested item retains `aria-current="page"`  
> And parent reflects expanded state via `aria-expanded="true"`

---

### ✅ Focus Management and Tab Behavior

**Scenario: Exit sidenavigation using Tab or Shift+Tab**  
> Given focus is anywhere inside sidenavigation  
> When I press Tab or Shift+Tab  

> > Then focus moves outside sidenavigation  

Reviewer: This is incorrect.
The focus moves between menubar's within the sidenavigation.

I suggest you to elaborate the "given" statement to say where the focus is and then write the rest of the steps accordingly.

> And all open submenus close

**Scenario: Close submenu using Escape**  

> > Given a submenu is open  

Reviewer: ```suggestion
> Given a submenu is open  and the focus is on one of the navigation items
```

> Then announces role `navigation` along with its accessible name (from `aria-label` or `aria-labelledby`)

**Scenario: Announce menubar structure**  
> Given screen reader navigates into menubar inside sidenavigation  
> Then announces role `menubar` with vertical orientation  
> And as user moves, each navigation item's label and role `menuitem` are announced  
> And submenu popup states are announced (`aria-haspopup`, `aria-expanded`)

**Scenario: Announce active page**  
> Given screen reader focus is on active navigation item  
> Then announces its label  
> And announces "current page" from `aria-current="page"`

**Scenario: Announce disabled state**  
> Given screen reader focus lands on disabled navigation item  
> Then announces its label  
> And announces "disabled"

**Scenario: Announce expand/collapse button**  
> Given screen reader focus is on expand/collapse button  
> Then announces its accessible label  
> And announces current state as either "collapsed" or "expanded"

**Scenario: Announce badge counts**  
> Given a navigation item includes a badge count  
> Then screen reader announces its full label combined with badge count value

> ---

Reviewer: We need more scenarios explaining the screen reader interactions between sidenav menuitems and submenus.
